src/Core/src/Platform/iOS/ViewExtensions.cs

```diff
@@ -698,5 +698,124 @@ internal static void UpdateLayerBorder(this CoreAnimation.CALayer layer, IButton
 			if (stroke.CornerRadius >= 0)
 				layer.CornerRadius = stroke.CornerRadius;
 		}
+
+		internal static T? FindResponder<T>(this UIView view) where T : UIResponder
+		{
+			var nextResponder = view as UIResponder;
+			while (nextResponder is not null)
+			{
+				nextResponder = nextResponder.NextResponder;
+
+				if (nextResponder is T responder)
+					return responder;
+			}
+			return null;
+		}
+
+		// static field to hold value when propagating back out the recursion
+		static bool IsRtl = false;
```

Comment:
Is it not possible to make the a variable in the "root" FindNextView and then have a ref parameter in the other methods. Static fields with temporary state is often a cause of random issues if something goes wrong.

Not sure if this works, but thinking like this:

```cs
internal static UIView? FindNextView(this UIView view, UIView superView, Func<UIView, bool> isValidType)
{
  var isRtl = false;
  var nextField = superView.FindNextView(originalRect, null, isValidType, ref isRtl);
  return nextField;
}

static UIView? FindNextView(this UIView view, CGRect originalRect, UIView? currentBest, Func<UIView, bool> isValidType, ref bool isRtl)
{
  isRtl |= view.SemanticContentAttribute == UISemanticContentAttribute.ForceRightToLeft;
  // ...
      if (TryFindNewBestView(originalRect, currentBest, child, out var newBest, ref isRtl))
  // ...
  return currentBest;
}

static bool TryFindNewBestView(CGRect originalRect, UIView? currentBest, UIView newView, out UIView newBest, ref bool isRtl)
{
  // ...
}
```

Also, this appears to not support a stack, so if I go into a RTL and then back out, the value is permanently set? I am thinking of a layout like this:

```xaml
<Stack IsRtl="false">
  <Stack IsRtl="true">
    <Entry/>
    <Entry/>
  </Stack>
  <Stack>
    <Entry/>
    <Entry/>
  </Stack>
</Stack>
```

Reply:
Yeah this is something that has been rough haha. I'm not too familiar with the UISemanticContentAttribute.ForceRightToLeft and did not think about the above situation. The issue I was having with the IsRtl lied in propogating back up from the recursive FindNextView and how the TryFindNewBestView compares the views based on the IsRtl. I will keep looking at this and perhaps need to change some of the logic of how I compare the NewBestView when coming back out the recursion

Reply:
If it is never un-set, does this mean the next time around it starts as RTL?

Reply:
The ref worked well though for keeping the same functionality though! Thanks!

Reply:
@mattleibow things are now being sorted by Logical order and I posted a video below showing that it handles RightToLeft and LeftToRight well now. I also added some unit tests regarding that as well!